PWGHF/D2H/TableProducer/candidateCreatorLbReduced.cxx

```diff
@@ -0,0 +1,339 @@
+    }
+    // invMassWindowLcPiTolerance is used to apply a slightly tighter cut than in LcPi pair preselection
+    // to avoid accepting LcPi pairs that were not formed in LcPi pair creator
+    float invMass2LcPiMin = (massLb - myInvMassWindowLcPi + invMassWindowLcPiTolerance) * (massLb - myInvMassWindowLcPi + invMassWindowLcPiTolerance);
+    float invMass2LcPiMax = (massLb + myInvMassWindowLcPi - invMassWindowLcPiTolerance) * (massLb + myInvMassWindowLcPi - invMassWindowLcPiTolerance);
+
+    for (const auto& collisionCounter : collisionsCounter) {
+      registry.fill(HIST("hEvents"), 1, collisionCounter.originalCollisionCount());
+    }
+
+    static int ncol = 0;
+    for (const auto& collision : collisions) {
+      auto thisCollId = collision.globalIndex();
+      auto candsLcThisColl = candsLc.sliceBy(candsLcPerCollision, thisCollId);
+      auto tracksPionThisCollision = tracksPion.sliceBy(tracksPionPerCollision, thisCollId);
+      runCandidateCreation<false>(collision, candsLcThisColl, tracksPionThisCollision, invMass2LcPiMin, invMass2LcPiMax);
+      if (ncol % 10000 == 0) {
+        LOGP(debug, "collisions parsed {}", ncol);
+      }
+      ncol++;
+    }
+  } // processData
+
+  PROCESS_SWITCH(HfCandidateCreatorLbReduced, processData, "Process data without any ML score", true);
+
+  void processDataWithLcMl(HfRedCollisionsWithExtras const& collisions,
+                           soa::Join<aod::HfRed3Prongs, aod::HfRed3ProngsCov, aod::HfRed3ProngsMl> const& candsLc,
+                           soa::Join<aod::HfRedTrackBases, aod::HfRedTracksCov> const& tracksPion,
+                           aod::HfOrigColCounts const& collisionsCounter,
+                           aod::HfCandLbConfigs const& configs)
+  {
+    // LcPi invariant-mass window cut
+    for (const auto& config : configs) {
+      myInvMassWindowLcPi = config.myInvMassWindowLcPi();
+    }
+    // invMassWindowLcPiTolerance is used to apply a slightly tighter cut than in LcPi pair preselection
+    // to avoid accepting LcPi pairs that were not formed in LcPi pair creator
+    float invMass2LcPiMin = (massLb - myInvMassWindowLcPi + invMassWindowLcPiTolerance) * (massLb - myInvMassWindowLcPi + invMassWindowLcPiTolerance);
+    float invMass2LcPiMax = (massLb + myInvMassWindowLcPi - invMassWindowLcPiTolerance) * (massLb + myInvMassWindowLcPi - invMassWindowLcPiTolerance);
+
+    for (const auto& collisionCounter : collisionsCounter) {
+      registry.fill(HIST("hEvents"), 1, collisionCounter.originalCollisionCount());
+    }
+
+    static int ncol = 0;
+    for (const auto& collision : collisions) {
+      auto thisCollId = collision.globalIndex();
+      auto candsLcThisColl = candsLc.sliceBy(candsLcPerCollision, thisCollId);
+      auto tracksPionThisCollision = tracksPion.sliceBy(tracksPionPerCollision, thisCollId);
+      runCandidateCreation<true>(collision, candsLcThisColl, tracksPionThisCollision, invMass2LcPiMin, invMass2LcPiMax);
```

Comment:
Is there a difference w.r.t. `processData` other than `true` on this line? Looks like too much unnecessary duplication to me.